### PR TITLE
Use different step sizes for different orders of derivatives

### LINKED ISSUE #607: The implementation of finite differences throws away its advantage over AD
I have heard quite a few times that finite differences are better than automatic differentiation in computing higher-order derivatives. However, this is not shown in this package. It can be seen in two examples in the documentation:
https://neuralpde.sciml.ai/stable/examples/3rd/
https://neuralpde.sciml.ai/stable/examples/ks/

that finite differences are quite bad. 

#445 was closed without actually fixing the issue. (Why are there some attached images from nowhere in the docs?). The culprit is the use of the same finite difference step size for all orders of derivatives, and the rounding error dominates when the order gets higher.

### PR DESCRIPTION
**Pull request by YichengDWu**
Closes #607 and #445

### CODE REVIEW

**Comment by YichengDWu** [comment]
The Kuramoto–Sivashinsky equation is finally working.
plot_10

File: src/symbolic_utilities.jl [hunk 1]
```diff
@@ -80,8 +80,8 @@ function transform_expression(pinnrep::PINNRepresentation, ex; is_integral = fal
     return ex
 end
 
-function get_ε(dim, der_num, eltypeθ)
-    epsilon = cbrt(eps(eltypeθ))
+function get_ε(dim, der_num, eltypeθ, order)
+    epsilon = ^(eps(eltypeθ), one(eltypeθ) / (2 + order))
```
Comment (ChrisRackauckas) [inline src/symbolic_utilities.jl:84]: This seems off by one? It should be `sqrt(eps)` at first order.

File: src/symbolic_utilities.jl [hunk 2]
```diff
@@ -80,8 +80,8 @@ function transform_expression(pinnrep::PINNRepresentation, ex; is_integral = fal
     return ex
 end
 
-function get_ε(dim, der_num, eltypeθ)
-    epsilon = cbrt(eps(eltypeθ))
+function get_ε(dim, der_num, eltypeθ, order)
```
Comment (ChrisRackauckas) [inline src/symbolic_utilities.jl:83]:
```suggestion
function get_ε(dim, der_num, ::Type{eltypeθ}, order) where eltypeθ
```

Comment (ChrisRackauckas) [reply src/symbolic_utilities.jl:83]: In fact, we should lock some of these down to just `Int` too

File: src/symbolic_utilities.jl (hunk 1 above)
Comment (YichengDWu) [reply src/symbolic_utilities.jl:84]: It was cbrt(eps). When order = 1 they are consistent.

Comment (ChrisRackauckas) [reply src/symbolic_utilities.jl:84]: oh yes, double checked the formulae. It's +2 for central differences like in https://github.com/JuliaDiff/FiniteDiff.jl/blob/master/src/epsilons.jl#L30

**Pull request merged**